apiserver/pkg/printers/projectcalico/printers.go

```diff
@@ -0,0 +1,169 @@
+	}
+	h.TableHandler(calicoNodeStatusColumnDefinitions, printCalicoNodeStatusList)
+	h.TableHandler(calicoNodeStatusColumnDefinitions, printCalicoNodeStatus)
+}
+
+func printCalicoNodeStatusList(statusList *calico.CalicoNodeStatusList, options printers.GenerateOptions) ([]metav1.TableRow, error) {
+	rows := make([]metav1.TableRow, 0, len(statusList.Items))
+	for i := range statusList.Items {
+		r, err := printCalicoNodeStatus(&statusList.Items[i], options)
+		if err != nil {
+			return nil, err
+		}
+		rows = append(rows, r...)
+	}
+	return rows, nil
+}
+
+func printCalicoNodeStatus(status *calico.CalicoNodeStatus, options printers.GenerateOptions) ([]metav1.TableRow, error) {
+	row := metav1.TableRow{
+		Object: runtime.RawExtension{Object: status},
+	}
+
+	// Define a function return true if the status contains information about a class.
+	hasClass := func(c calico.NodeStatusClassType) bool {
+		for _, class := range status.Spec.Classes {
+			if class == c {
+				return true
+			}
+		}
+		return false
+	}
+
+	var classes string
+	for _, class := range status.Spec.Classes {
+		classes += fmt.Sprintf("%s,", class)
+	}
+	classesStr := fmt.Sprintf("[%s]", strings.TrimSuffix(classes, ","))
+
+	var lastUpdatedStr string
+	lastUpdatedDate := status.Status.LastUpdated
+	if !lastUpdatedDate.IsZero() {
+		lastUpdatedStr = fmt.Sprintf("%s ago", translateTimestampSince(lastUpdatedDate))
+	}
+
+	row.Cells = append(row.Cells,
+		status.Name,
+		status.Spec.Node,
+		classesStr,
+		fmt.Sprintf("%ds", *status.Spec.UpdatePeriodSeconds),
+		translateTimestampSince(status.CreationTimestamp),
+		lastUpdatedStr)
+
+	if options.Wide {
+		var agentStateStr string
+		if hasClass(calico.NodeStatusClassTypeAgent) {
+			agent := status.Status.Agent
+			if !reflect.ValueOf(agent.BIRDV4).IsZero() {
+				agentStateStr = fmt.Sprintf("v4(%s) ", agent.BIRDV4.State)
+			}
+			if !reflect.ValueOf(agent.BIRDV6).IsZero() {
+				agentStateStr += fmt.Sprintf("v6(%s)", agent.BIRDV6.State)
+			}
+		}
+
+		var peersStr string
+		if hasClass(calico.NodeStatusClassTypeBGP) {
+			bgp := status.Status.BGP
+			if !reflect.ValueOf(bgp.PeersV4).IsZero() {
+				peersStr = fmt.Sprintf("v4(%d/%d/%d) ",
+					len(bgp.PeersV4), bgp.NumberEstablishedV4, bgp.NumberNotEstablishedV4)
+			}
+			if !reflect.ValueOf(bgp.PeersV6).IsZero() {
+				peersStr += fmt.Sprintf("v6(%d/%d/%d) ",
+					len(bgp.PeersV6), bgp.NumberEstablishedV6, bgp.NumberNotEstablishedV6)
+			}
+		}
+
+		getNumberOfBGPRoutes := func(routes []calico.CalicoNodeRoute) int {
+			n := 0
+			for _, r := range routes {
+				if (r.LearnedFrom.SourceType == calico.RouteSourceTypeBGPPeer) ||
+					(r.LearnedFrom.SourceType == calico.RouteSourceTypeNodeMesh) {
+					n++
+				}
+			}
+			return n
+		}
+
+		var routesStr string
+		if hasClass(calico.NodeStatusClassTypeRoutes) {
+			routes := status.Status.Routes
+			if !reflect.ValueOf(routes.RoutesV4).IsZero() {
+				routesStr = fmt.Sprintf("v4(%d) ", getNumberOfBGPRoutes(routes.RoutesV4))
```

Comment:
Same here - I think additional column headings are more readable than putting multiple bits of information in a single column.

So, maybe columns like "NUM-V4-ROUTES and NUM-V6-ROUTES"